ultralytics/solutions/dwell_time_analyzer.py

```diff
@@ -0,0 +1,475 @@
+                del self.object_current_zone[track_id]
+
+    def compute_advanced_analytics(
+        self,
+    ) -> Tuple[Optional[float], int, int, Dict[str, float]]:
+        """
+        Computes funnel conversion rates and average dwell times per zone.
+
+        Returns:
+            Tuple[Optional[float], int, int, Dict[str, float]]:
+                - Funnel conversion rate as a percentage.
+                - Number of objects that visited the first funnel stage.
+                - Number of objects that completed all funnel stages.
+                - Dictionary mapping zone names to their average dwell times.
+        """
+        funnel_conversion: Optional[float] = None
+        visited_first: int = 0
+        visited_all_stages: int = 0
+        avg_dwell_times: Dict[str, float] = {}
+
+        if self.enable_funnel and self.funnel_stages and len(self.funnel_stages) >= 2:
+            funnel_stages = self.funnel_stages
+        else:
+            funnel_stages = None
+
+        zone_dwell_times: defaultdict = defaultdict(float)
+        zone_visit_counts: defaultdict = defaultdict(int)
+
+        for track_id, seq in self.object_zone_sequence.items():
+            zones_visited = [z[0] for z in seq]
+
+            # Funnel logic for multiple stages
+            if funnel_stages:
+                # Check if object visited the first stage
+                if funnel_stages[0] in zones_visited:
+                    visited_first += 1
+                    # Check if all stages are visited in order
+                    current_index = -1
+                    all_stages_found = True
+                    for stage in funnel_stages:
+                        try:
+                            next_index = zones_visited.index(stage, current_index + 1)
+                            current_index = next_index
+                        except ValueError:
+                            all_stages_found = False
+                            break
+                    if all_stages_found:
+                        visited_all_stages += 1
+
+            # Average dwell time if enabled
+            if self.enable_avg_dwell:
+                for z_name, e_time, x_time in seq:
+                    dwell = x_time - e_time
+                    zone_dwell_times[z_name] += dwell
+                    zone_visit_counts[z_name] += 1
+
+        # Compute funnel conversion rate if funnel enabled
+        if funnel_stages and visited_first > 0:
+            funnel_conversion = (visited_all_stages / visited_first) * 100.0
+
+        # Compute average dwell times if enabled
+        if self.enable_avg_dwell:
+            for z_name, total_time in zone_dwell_times.items():
+                avg = total_time / zone_visit_counts[z_name] if zone_visit_counts[z_name] > 0 else 0
+                avg_dwell_times[z_name] = avg
+
+        return funnel_conversion, visited_first, visited_all_stages, avg_dwell_times
+
+    def display_analytics(self, im0: Any) -> None:
```

Comment:
@Muhammad-237 we already have the [`display_analytics`](https://github.com/ultralytics/ultralytics/blob/a3d807be13af2d8fa008f522020411d2d37e4860/ultralytics/utils/plotting.py#L657) function, can you please use that? Thank you!

Reply:
i have reviewed it but their is a big difference in these two desplay_analytics function

in my display analytics function their are lot's of dynamic analytics processing related to funnel and time processing and dwell time averages.